xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1164,6 +1182,53 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     return true;
   }
 
+  absl::Status F8ScaleD(HloInstruction* instr, HloInstruction* existing_gemm,
+                        HloInstruction* d_scale) {
+    if (!ShapeUtil::IsScalar(d_scale->shape())) {
+      return OkStatus();
+    }
+
+    TF_ASSIGN_OR_RETURN(auto gpu_backend_config,
+                        existing_gemm->backend_config<GpuBackendConfig>());
+    const GemmBackendConfig& gemm_backend_config =
+        gpu_backend_config.gemm_backend_config();
+
+    // When the result of an FP8 GEMM is scaled but not type converted to FP8,
+    // the scaling factor must be forwarded to the Custom Call as a_scale
+    // (chosen here) or b_scale. The scaling factor is fused here when no input
+    // scaling factors were fused during the creation of the Custom Call. When
+    // the maximum of the absolute value of the result of an FP8 GEMM is
+    // calculated and the result is scaled and type converted to FP8, the
+    // scaling of the result is fused in F8ConvertD.
+    const int a_scale_operand_idx = gemm_backend_config.beta() == 0.0 ? 2 : 3;
```

Comment:
If beta is nonzero, we cannot fuse d_scale at all. Since if the cublas call includes a matrix addition, scaling the output is not equivalent to scaling the input.

I believe fusing d_sclae as a_scale is also not valid when there are certain activations, like gelu (relu is fine I think).

Add a test for the case where the output is non-FP8, scaled, and there is a matrix addition.